File: back/DevC/src/main/java/com/back/devc/domain/post/comment/dto/CommentListResponse.java
package com.back.devc.domain.post.comment.dto;

import java.util.List;

public record CommentListResponse(
        List<CommentResponse> comments,
        int page,
        int size,
        long totalElements,
        int totalPages,
        boolean hasNext
) {
    public CommentListResponse(List<CommentResponse> comments) {
        this(comments, 0, comments.size(), comments.size(), 1, false);
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/comment/dto/CommentListResponse.kt
package com.back.devc.domain.post.comment.dto

data class CommentListResponse(
    val comments: List<CommentResponse>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
    val hasNext: Boolean,
) {
    constructor(comments: List<CommentResponse>) : this(
        comments = comments,
        page = 0,
        size = comments.size,
        totalElements = comments.size.toLong(),
        totalPages = 1,
        hasNext = false,
    )
}